base/stat.jl

```diff
@@ -57,7 +59,67 @@ StatStruct(buf::Union{Vector{UInt8},Ptr{UInt8}}) = StatStruct(
     ccall(:jl_stat_ctime,   Float64, (Ptr{UInt8},), buf),
 )
 
-show(io::IO, st::StatStruct) = print(io, "StatStruct(mode=0o$(string(filemode(st), base = 8, pad = 6)), size=$(filesize(st)))")
+function show(io::IO, st::StatStruct)
+    compact = get(io, :compact, true)
+    function iso_datetime_with_relative(t, tnow)
+        sprint() do iob2
+            print(iob2, Libc.strftime("%FT%T%z", t))
+            secdiff = t - tnow
+            for (d, name) in ((24*60*60,"day"),(60*60,"hour"),(60,"minute"),(1,"second"))
+                tdiff = round(Int, div(abs(secdiff), d))
+                (tdiff == 0 && name != "second") && continue # find first unit difference
+                if tdiff == 0 && name == "second"
+                    print(iob2, " (just now)")
+                    break
+                end
+                plural = tdiff == 1 ? "" : "s"
+                when = secdiff < 0 ? "ago" : "in the future"
+                print(iob2, " ($(tdiff) $(name)$(plural) $(when))")
+                break
+            end
+        end
+    end
+    str = if compact
+        sprint() do iob
+            print(iob, "$(st.desc)")
+            print(iob, " size: $(st.size) bytes")
+            print(iob, " device: $(st.device)")
+            print(iob, " inode: $(st.inode)")
+            print(iob, " mode: 0o$(string(filemode(st), base = 8, pad = 6)) ($(filemode_string(st)))")
+            print(iob, " nlink: $(st.nlink)")
+            username = Base.getusername(st.uid)
+            isnothing(username) ? print(iob, " uid: $(st.uid)") : print(iob, " uid: $(st.uid) ($username)")
+            groupname = Base.getgroupname(st.gid)
+            isnothing(groupname) ? print(iob, " gid: $(st.gid)") : print(iob, " gid: $(st.gid) ($groupname)")
+            print(iob, " rdev: $(st.rdev)")
+            print(iob, " blksize: $(st.blksize)")
+            print(iob, " blocks: $(st.blocks)")
+            tnow = Libc.TimeVal().sec
+            print(iob, " mtime: $(iso_datetime_with_relative(st.mtime, tnow))")
+            println(iob, " ctime: $(iso_datetime_with_relative(st.ctime, tnow))")
+        end
+    else
+        sprint() do iob
```

Comment:
This should be the code for a different function (instead of compact):
```suggestion
function show(io::IO, ::MIME"text/plain", st::StatStruct)
```

Reply:
You mean that you wouldn't have the user select compact or not via the compact flag?

Reply:
Tests are passing now, and I believe this is the only outstanding issue, but I need help understanding what you mean. Are you suggesting the user selects the compact or full versions via some other mechanism than setting `julia> Base.active_repl.options.iocontext[:compact] = false` ?
Setting `compact` via that method is rather laborious, so I'm definitely not attached to it.